bindata/etcd/cluster-restore.sh

```diff
@@ -84,28 +109,31 @@ done
 # wait for every static pod container to stop
 wait_for_containers_to_stop "${STATIC_POD_CONTAINERS[@]}"
 
-if [ ! -d ${ETCD_DATA_DIR_BACKUP} ]; then
-  mkdir -p ${ETCD_DATA_DIR_BACKUP}
+if [ ! -d "${ETCD_DATA_DIR_BACKUP}" ]; then
+  mkdir -p "${ETCD_DATA_DIR_BACKUP}"
 fi
 
 # backup old data-dir
 if [ -d "${ETCD_DATA_DIR}/member" ]; then
   if [ -d "${ETCD_DATA_DIR_BACKUP}/member" ]; then
     echo "removing previous backup ${ETCD_DATA_DIR_BACKUP}/member"
-    rm -rf ${ETCD_DATA_DIR_BACKUP}/member
+    rm -rf "${ETCD_DATA_DIR_BACKUP}"/member
   fi
   echo "Moving etcd data-dir ${ETCD_DATA_DIR}/member to ${ETCD_DATA_DIR_BACKUP}"
-  mv ${ETCD_DATA_DIR}/member ${ETCD_DATA_DIR_BACKUP}/
+  mv "${ETCD_DATA_DIR}"/member "${ETCD_DATA_DIR_BACKUP}"/
 fi
 
 # Restore static pod resources
-tar -C ${CONFIG_FILE_DIR} -xzf ${BACKUP_FILE} static-pod-resources
+tar -C "${CONFIG_FILE_DIR}" -xzf "${BACKUP_FILE}" static-pod-resources
 
 # Copy snapshot to backupdir
-cp -p ${SNAPSHOT_FILE} ${ETCD_DATA_DIR_BACKUP}/snapshot.db
+cp -p "${SNAPSHOT_FILE}" "${ETCD_DATA_DIR_BACKUP}"/snapshot.db
 
-echo "starting restore-etcd static pod"
-cp -p ${RESTORE_ETCD_POD_YAML} ${MANIFEST_DIR}/etcd-pod.yaml
+# Extract and start restore-etcd pod.yaml
+if ! extract_and_start_restore_etcd_pod; then
+  # If backup doesn't have restore-pod.yaml, copy it from local disk
+  cp -p ${RESTORE_ETCD_POD_YAML} ${MANIFEST_DIR}/etcd-pod.yaml
```

Comment:
Would the error be clear to the user if this file also didn't exist? Do we need some sanity precheck to ensure the backup is complete?

Reply:
based on conversation with @retroflexer probability of invalid backup would be low so no need to cover it outside of cp stderr.

Reply:
I think the error would be self-explanatory. If the backup doesn't have the restore pod, and local disk doesn't have as well, then the customer has some other issues, any way.